integration/networking/bridge_test.go

```diff
@@ -162,15 +169,19 @@ func TestBridgeICC(t *testing.T) {
 
 			pingHost := tc.pingHost
 			if pingHost == "" {
-				if tc.linkLocal {
+				if tc.isLinkLocal {
 					inspect := container.Inspect(ctx, t, c, id1)
 					pingHost = inspect.NetworkSettings.Networks[bridgeName].GlobalIPv6Address + "%eth0"
 				} else {
 					pingHost = ctr1Name
 				}
 			}
 
-			pingCmd := []string{"ping", "-c1", "-W3", pingHost}
+			pingCmd := []string{"ping", "-c1", "-W3"}
+			if tc.isIPv6 {
+				pingCmd = append(pingCmd, "-6")
+			}
+			pingCmd = append(pingCmd, pingHost)
```

Comment:
Is the reverse also the case? I.e., if it's not IPv6, do we need to add `-4` ?

Reply:
^^ in that case, we could consider just defining 2 "full" commands, instead of `append` (might be more readable to have the whole command at a glance)

Reply:
I think the current IPv4 tests don't have IPv6 addresses, so ping has to use v4. But, perhaps we should also test that "ping -4" works on a network with IPv6?

Rather than two full commands, maybe something like this to include in a single command? ...

```
ipv := "-4"
if tc.isIPv6 {
  ipv = "-6"
}
```

Reply:
> Rather than two full commands, maybe something like this to include in a single command? ...

Yeah, in this case, either way could be fine; it's all still fairly easy to read.

I've had cases though where we were constructing the command to run in a "too complicated" way, and now it was needed to run 10s of lines of code to understand "what command are we running here??". So where possible (and where it makes sense), having a single like `sh -c 'foo bar --baz --opt foo --hello'` can be useful, and can help spotting bugs/omissions in the code.

Reply:
> Is the reverse also the case? I.e., if it's not IPv6, do we need to add -4 ?

We don't need to specify `-4` as it's the default.

> I think the current IPv4 tests don't have IPv6 addresses, so ping has to use v4. But, perhaps we should also test that "ping -4" works on a network with IPv6?

I'll keep that for a follow-up. I prefer to keep this PR tiny to make sure https://github.com/moby/moby/pull/46609 can be merged soon.